Makefile

```diff
@@ -8,6 +8,12 @@ lint:
 test: lint
 	CGO_ENABLED=0 go test -timeout 1m -v -covermode=atomic ./...
 
+## Generate mocks
+gen-mock:
```

Comment:
Can we use go-mock instead? We use that in other places.

Reply:
I found mockery favourable for a few reasons over gomock,
- `mockery` mocks are supported by the testing library of our choice - [github.com/testify/assert/mock](https://github.com/stretchr/testify?tab=readme-ov-file#mock-package) this way we don't require `gomock` to be imported.
- It supports a central config and is more explicit on which interfaces we want mocks for, and doesn't require `// go:generate` annotations spread across files 
- The errors returned by failing tests using [github.com/testify/assert/mock](https://github.com/stretchr/testify?tab=readme-ov-file#mock-package) are easier to follow compared to gomock